app/components/graph-editor/drawable-interfaces.ts

```diff
@@ -0,0 +1,398 @@
+     */
+    replaceEdge(original: DrawableEdge, replacement: DrawableEdge): void;
+
+    /**
+     * removeEdge  
+     *   Guarantees that the given edge is not present in the graph.
+     */
+    removeEdge(edge: DrawableEdge): void;
+
+}
+
+/**
+ * DrawableEdge  
+ *   Interface that exposes drawable edge properties.
+ */
+export interface DrawableEdge {
+
+    /**
+     * source  
+     *   The source node of the edge.
+     */
+    source: DrawableNode | null;
+
+    /**
+     * destination  
+     *   The destination node of the edge.
+     */
+    destination: DrawableNode | null;
+
+    /**
+     * showSourceArrow  
+     *   True to draw an arrow pointing to the source node; otherwise, false.
+     */
+    showSourceArrow: boolean;
+
+    /**
+     * showDestinationArrow  
+     *   True to draw an arrow pointing to the destination node; otherwise, false.
+     */
+    showDestinationArrow: boolean;
+
+    /**
+     * color  
+     *   The color of the edge. This can be any valid `CSS` color string.
+     */
+    color: string;
+
+    /**
+     * lineStyle  
+     *   The line style of the edge. This can be `solid`, `dotted`, or `dashed`.
+     */
+    lineStyle: string;
+
+    /**
+     * lineWidth  
+     *   The width of the edge. This value must be non-negative.
+     */
+    lineWidth: number;
+
+    /**
+     * label  
+     *   The text label to be displayed by the edge.
+     */
+    label: string;
+
+    // TODO:
+    // more display properties.
+}
+
+/**
+ * DrawableNode  
+ *   Interface that exposes drawable node properties.
+ */
+export interface DrawableNode {
+
+    /**
+     * x  
+     *   The x-coordinate of the center of the node.
+     */
+    x: number;
+
+    /**
+     * y  
+     *   The y-coordinate of the center of the node.
+     */
+    y: number;
+
+    /**
+     * label  
+     *   The text label to be displayed in the node.
+     */
+    label: string;
+
+    /**
+     * shape  
+     *   The shape of the node.
+     */
+    shape: string; // TODO: for now this only supports circles and squares.
+
+    /**
+     * color  
+     *   The background color of the node.  Can be any valid `CSS` color string.
+     */
+    color: string;
+
+    /**
+     * borderColor  
+     *   The border color of the node.  Can be any valid `CSS` color string.
+     */
+    borderColor: string;
+
+    /**
+     * borderStyle  
+     *   The line style of the border. Can be `solid`, `dotted`, or `dashed`.
+     */
+    borderStyle: string;
+
+    /**
+     * borderWidth  
+     *   The line width of the border. Set to 0 to draw no border; value must be
+     *   non-negative.
+     */
+    borderWidth: number;
+
+    // TODO:
+    // more display properties
+}
+
+// Type aliases ////////////////////////////////////////////////////////////////
+
+
+/**
+ * Drawable  
+ *   Type alias for the the union type of `DrawableEdge` and `DrawableNode`.
+ */
+export type Drawable = DrawableEdge | DrawableNode;
```

Comment:
Since you are using them as the same thing anyway for some functions, it might be worth making Drawable a superclass of DrawableEdge and DrawableNode.